reactor-core/src/main/java/reactor/core/publisher/MonoStreamCollector.java

```diff
@@ -164,7 +164,11 @@ public void onComplete() {
 				return;
 			}
 
-			complete(r);
+			if (r != null) {
+				complete(r);
+			} else {
+				actual.onError(Operators.onOperatorError(new NullPointerException("Collector returned null"), actual.currentContext()));
+			}
```

Comment:
WDYT about an early return?

```suggestion
			if (r == null) {
				actual.onError(Operators.onOperatorError(new NullPointerException("Collector returned null"), actual.currentContext()));
				return;
			}
			
			complete(r);
```